ccmlib/cluster.py

```diff
@@ -304,23 +301,19 @@ def start(self, no_wait=False, verbose=False, wait_for_binary_proto=False, wait_
             if not node.is_running():
                 raise NodeError("Error starting {0}.".format(node.name), p)
 
-        if not no_wait and self.cassandra_version() >= "0.8":
-            # 0.7 gossip messages seems less predictible that from 0.8 onwards and
-            # I don't care enough
-            for node, _, mark in started:
-                for other_node, _, _ in started:
-                    if other_node is not node:
-                        node.watch_log_for_alive(other_node, from_mark=mark)
-
-        if wait_other_notice:
-            for old_node, mark in marks:
-                for node, _, _ in started:
-                    if old_node is not node:
-                        old_node.watch_log_for_alive(node, from_mark=mark)
-
-        if wait_for_binary_proto:
-            for node, p, mark in started:
-                node.wait_for_binary_interface(process=p, verbose=verbose, from_mark=mark)
+        if not no_wait:
+            if wait_other_notice:
+                # For each node
+                for node, _, mark in started:
+                    # Go through every other node
+                    for other_node, _, _ in started:
+                        if other_node is not node:
```

Comment:
[`itertools.permutations`](https://docs.python.org/2/library/itertools.html#itertools.permutations) will handle this nested-loop-and-filter logic for you:

```python
>>> import itertools, pprint
>>> pprint.pprint(list(itertools.permutations('ABCD', 2)))
[('A', 'B'),
 ('A', 'C'),
 ('A', 'D'),
 ('B', 'A'),
 ('B', 'C'),
 ('B', 'D'),
 ('C', 'A'),
 ('C', 'B'),
 ('C', 'D'),
 ('D', 'A'),
 ('D', 'B'),
 ('D', 'C')]
```